### ISSUE TITLE
Multi Label Classifier sample

### ISSUE THREAD

**Issue by ronbutan** [dotnet/docs#5385]
I look forward to a sample illustrating Multiclass Classification model. Common example will be News categorization. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e855e630-1a30-d138-4ac6-bfeefebf9aae
* Version Independent ID: afc0a7d1-d3c2-acf2-d4ea-d2565e22b237
* Content: [ML.NET tutorials](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/#feedback)
* Content Source: [docs/machine-learning/tutorials/index.md](https://github.com/dotnet/docs/blob/master/docs/machine-learning/tutorials/index.md)
* Service: **dotnet-ml**
* GitHub Login: @JRAlexander
* Microsoft Alias: **johalex**

**Comment by mairaw** [comment]
@asthana86 @aditidugar please take a look at this request for a new tutorial for ML.NET.

**Comment by aditidugar** [comment]
Thanks for the feedback, @ronbutan! We're still working on building out all of our tutorials and samples, but a multi-classification sample will be added soon.

**Comment by mairaw** [comment]
Maybe we should add that link to this page then: https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/?

**Comment by aditidugar** [comment]
Yes, @brendankowitz is right, that's a multi-classification sample. @mairaw, we can certainly link from there temporarily, but it's not really a step-by-step tutorial like our other walkthroughs. We're working on creating a more end-to-end multi classification sample based on some of the demos we showed, and will have that ready soon.

**Comment by ronbutan** [comment]
Appreciate the speedy response.  I was wondering how to construct the class (i.e. IrisData) with multi labels (e.g. public string Label vs public string[] Labels). A simple code sample will definitely help.

**Comment by aditidugar** [comment]
Hey @ronbutan! Thanks for the clarification here, I think we may have misunderstood your earlier question. Here is some guidance that may help:

1. Multi-label classification: If you're looking to classify more than one label (predicting iris type in addition to petal size for example), then we unfortunately do not support this at the moment. The guidance here would be to train individual models for each of these, and then combine the predictions together. If you're interested in this support please open an issue on our [GitHub repo](https://github.com/dotnet/machinelearning).

2. Multi-class classification. If you are looking for an array output since there are multiple values for each label, the "Score" attribute can provide the array of scores (one per class). You can [check out a sample for that here](https://github.com/dotnet/machinelearning/blob/master/test/Microsoft.ML.Tests/Scenarios/IrisPlantClassificationTests.cs). We also have an [open issue](https://github.com/dotnet/machinelearning/issues/158) about this because you currently cannot get the string labels for the classes other than the top class. 

Hope this helps!

**Comment by Finansnet** [comment]
Hi

The sample with flowers looks and works awesome. I am impressed. Could you please avice me what kind of classifier should I use if one (or more) of my input features contains descreet values like color of the flower?

Thanks
Marek

**Comment by aditidugar** [comment]
Hi @Finansnet! If I understand your question correctly, you're looking to handle input features for the color of flower such as "yellow", "pink", "blue", etc.

You can use the CategoricalOneHotVectorizer() function to transform these into numeric values so you can then process them in your learning algorithm. Please take a look at the [Taxi Fare example](https://docs.microsoft.com/en-gb/dotnet/machine-learning/tutorials/taxi-fare) to see how this can be implemented.

**Comment by Finansnet** [comment]
Hi @aditidugar

That was exactly I was looking for. I have checked in on my training data and the results are very promising. Thank you.

Marek

**Comment by stephenstroud** [comment]
@aditidugar Any update on Multi-label classification support please?

**Comment by JRAlexander** [comment]
@stephenstroud - on the way soon! In the meantime, check out the sample here: https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/MulticlassClassification-GitHubLabeler

**Comment by slarrieu** [comment]
> @stephenstroud - on the way soon! In the meantime, check out the sample here: https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/MulticlassClassification-GitHubLabeler

Hi Alexander, sorry to bother but I've been looking for multi-label support/doc and can't find it anywhere. It is currently supported by ml.net?

**Comment by Lukaa0** [comment]
Any update on this?

**Comment by marchy** [comment]
@JRAlexander what's the update on multi-label classification?

It's been over a year now since the 'on the way soon' comment. Only examples are multi-class algorithms which produce only a single label.

We much need a solution for this – including scoring that accounts for the multi-label aspect of things (ie: whether the algorithm correctly labelled with the right number of tags)

**Comment by natke** [comment]
Hi @marchy, ML.NET currently does not have native support for multi-label classification. You could raise an issue over on the product GitHub repo if you like - it's also open source: https://github.com/dotnet/machinelearning. 

Your other option would be to use a pre-trained TensorFlow or ONNX model that already provided this functionality. You can see an example of how to load a pre-trained model here: https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/text-classification-tf

**Comment by hjoshuaj** [comment]
@JRAlexander @natke Since the last comment on this question was over a year ago, is there an update on whether ML.NET supports multi-label classification?

I'm currently pulling in my pre-trained TensorFlow model, but it would be nice to have everything under one roof.

**Comment by mairaw** [comment]
Pinging @briacht to see if she can provide an update

**Comment by TusharMalap** [comment]
> Pinging @briacht to see if she can provide an update

Hey is there way I can achieve hierarchical classification in ml.net?

**Comment by lockhartsoftware** [comment]
Does ML.Net support multi-label classification?
